File: recursion/ex1.cpp
#include <iostream>
#include <ostream>

using namespace std;
string reverseString(const string& str)
{
    if (str.length() <= 1) {
        return str;
    }
    else {
        return reverseString(str.substr(1)) + str[0];
    }
}
int main(){
    string input, reversed;
    cout << "Enter the string to reverse:" << endl;
    cin >> input;
    reversed = reverseString(input);
    cout <<"Reversed string: "<< reversed << endl;
    return 0;
}

File: recursion/ex1.cs
using System;

class Program
{
    static void ReverseString(string str)
    {
        if (str.Length == 0)
            return;

        ReverseString(str.Substring(1));
        Console.Write(str[0]);
    }

    static void Main()
    {
        string input = "tiger";
        ReverseString(input);
    }
}

